core/modules/replica/FindRequest.h

```diff
@@ -0,0 +1,196 @@
+     * low-level pointers).
+     *
+     * @param serviceProvider  - a host of services for various communications
+     * @param worker           - the identifier of a worker node (the one where the chunk is
+     *                           expected to be located) at a destination of the chunk
+     * @param database         - the name of a database
+     * @param chunk            - the number of a chunk to find (implies all relevant tables)
+     * @param onFinish         - an optional callback function to be called upon a completion of
+     *                           the request.
+     * @param priority         - a priority level of the request
+     * @param computeCheckSum  - tell a worker server to compute check/control sum on each file
+     * @param keepTracking     - keep tracking the request before it finishes or fails
+     * @param messenger        - an interface for communicating with workers
+     */
+    static pointer create(ServiceProvider::pointer const& serviceProvider,
+                          boost::asio::io_service& io_service,
+                          std::string const& worker,
+                          std::string const& database,
+                          unsigned int  chunk,
+                          callback_type onFinish,
+                          int  priority,
+                          bool computeCheckSum,
+                          bool keepTracking,
+                          std::shared_ptr<Messenger> const& messenger);
+
+private:
+
+    /**
+     * Construct the request with the pointer to the services provider.
+     */
+    FindRequestM(ServiceProvider::pointer const& serviceProvider,
+                 boost::asio::io_service& io_service,
+                 std::string const& worker,
+                 std::string const& database,
+                 unsigned int  chunk,
+                 callback_type onFinish,
+                 int  priority,
+                 bool computeCheckSum,
+                 bool keepTracking,
+                 std::shared_ptr<Messenger> const& messenger);
+
+    /**
+      * Implement the method declared in the base class
+      *
+      * @see Request::startImpl()
+      */
+    void startImpl() final;
+
+    /// Start the timer before attempting the previously failed
+    /// or successfull (if a status check is needed) step.
+    void wait();
+
+    /// Callback handler for the asynchronious operation
+    void awaken(boost::system::error_code const& ec);
+
+    /// Send the serialized content of the buffer to a worker
+    void send();
+
+    /// Process the completion of the requested operation
+    void analyze(bool success,
+                lsst::qserv::proto::ReplicationResponseFind const& message);
+
+    /**
+     * Notifying a party which initiated the request.
+     *
+     * This method implements the corresponing virtual method defined
+     * bu the base class.
+     */
+    void notify() final;
+
+private:
+
+    // Parameters of the object
```

Comment:
unnecessary comment?

Reply:
Fixed. Though, my original intent was to separate a group of parameters passed into a class's constructor from other member variables (cache). Later I mostly gave up on this due to a very limited benefits of this.